src/main/java/com/rocketden/main/service/ProblemService.java

```diff
@@ -83,6 +86,67 @@ public ProblemDto getProblem(String problemId) {
         return ProblemMapper.toDto(problem);
     }
 
+    public ProblemDto editProblem(String problemId, ProblemDto updatedProblem) {
+        Problem problem = repository.findProblemByProblemId(problemId);
+
+        if (problem == null) {
+            throw new ApiException(ProblemError.NOT_FOUND);
+        }
+
+        if (updatedProblem == null || updatedProblem.getName() == null
```

Comment:
Add check for `updatedProblem.getTestCases() == null` as well, otherwise a 405 method not allowed error is shown, instead of `ProblemError.EMPTY_FIELD`.

OR, if you want users to pass in a `null` field for `testCases` to indicate that the updated problem should have no test cases, you should modify the for loop line `for (ProblemTestCaseDto testCaseDto : updatedProblem.getTestCases()) {` so that it doesn't throw an error.

Reply:
Went with option one 👍